pandas/core/dtypes/cast.py

```diff
@@ -246,6 +246,85 @@ def trans(x):
     return result
 
 
+def maybe_cast_result(result, obj, numeric_only: bool = False, how: str = ""):
+    """
+    Try to cast the result to the desired type,
+    we may have roundtripped through object in the mean-time.
+
+    If numeric_only is True, then only try to cast numerics
+    and not datetimelikes.
+
+    """
+    if obj.ndim > 1:
+        dtype = obj._values.dtype
+    else:
+        dtype = obj.dtype
+    dtype = maybe_cast_result_dtype(dtype, how)
+
+    if not is_scalar(result):
+        if is_extension_array_dtype(dtype) and dtype.kind != "M":
+            # The function can return something of any type, so check
+            #  if the type is compatible with the calling EA.
+            # datetime64tz is handled correctly in agg_series,
+            #  so is excluded here.
+
+            if len(result) and isinstance(result[0], dtype.type):
+                cls = dtype.construct_array_type()
+                result = try_cast_to_ea(cls, result, dtype=dtype)
+
+        elif numeric_only and is_numeric_dtype(dtype) or not numeric_only:
+            result = maybe_downcast_to_dtype(result, dtype)
+
+    return result
+
+
+def maybe_cast_result_dtype(dtype, how):
```

Comment:
if you can type things, pls do it. dtype: DtypeObj, how: str -> DtypeObj

Reply:
For maybe_cast_result, currently the only code paths have obj being a series, so I typed as ABCSeries. Was not 100% confident that's appropriate, maybe it should be left more general?